### DISCUSSION TITLE
Can't create a new discussion

### DISCUSSION THREAD

**Discussion by softwarecreations** [community/community#51560] [Apps, API and Webhooks]
### Select Topic Area

Bug

### Body

Thanks for making this API available.
I've successfully used the API's to perform some actions on my account via personal access tokens.

# Can't create discussion on public repo via API
Discussions are already enabled on my repo.
I run this command (both ChatGPT and Bing Chat gave me the same endpoint to perform this task)
```
curl -X POST -H "Accept: application/vnd.github.v3+json" -H "Authorization: token MY_PERSONAL_ACCESS_TOKEN" https://api.github.com/repos/softwarecreations/a-foo/discussions -d '{"title":"Test Title","body":"Test Body"}'
```

I have tried this using both classic and fine-grained personal access tokens that I ticked every single possible permission checkbox before creating.
Regardless I get the error
```
{
  "message": "Not Found",
  "documentation_url": "https://docs.github.com/rest"
}
```

I can't find any documentation for the discussions Github REST endpoint in Github's documentation. The closest thing I can find is TEAMS discussions, which I assume is not the same thing as public discussions on a public repo?

**Comment by Gisgar3** [comment]
Hey there, welcome to the **GitHub Community Forum**! As far as I'm aware, GitHub's REST API has no endpoints for interacting with Discussions. However, their GraphQL API does! If you're interested in using that, I provided some code below to query the repository's ID and its correlating Discussion category IDs (which are required to create a mutation through the GraphQL API):

```graphql
# Get the repository ID and Discussion category IDs (names outputted to better identify each category)
query {
  repository(name: "<sample-repo>", owner: "<sample-owner>") {
    id
    discussionCategories (first: 50) {
      pageInfo {
        hasPreviousPage
        hasNextPage
        endCursor
        startCursor
      }
      edges {
        node {
          id
          name
        }
      }
    }
  }
}
```
```graphql
# Create the discussion using the defined repository ID and category ID (body and title parameters are user-preference)
mutation {
  createDiscussion(input: {repositoryId: "1234", categoryId: "5678", body: "The body", title: "The title"}) {
    # Response from mutation containing newly created Discussion information
    discussion {
      id
    }
  }
}
```

I hope this helps you out! I wish GitHub's REST API had endpoints for Discussions too, but luckily they provide some through GraphQL 👍🏻 :octocat:
